components/proxy/src/test/kotlin/com/hotels/styx/proxy/backends/file/FileBackedBackendServicesRegistryTest.kt

```diff
@@ -0,0 +1,217 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.proxy.backends.file
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.hotels.styx.api.extension.Origin
+import com.hotels.styx.api.extension.Origin.newOriginBuilder
+import com.hotels.styx.api.extension.service.BackendService
+import com.hotels.styx.api.extension.service.ConnectionPoolSettings
+import com.hotels.styx.api.extension.service.HealthCheckConfig.newHealthCheckConfigBuilder
+import com.hotels.styx.api.extension.service.StickySessionConfig
+import com.hotels.styx.api.extension.service.spi.Registry
+import com.hotels.styx.applications.BackendServices.newBackendServices
+import com.hotels.styx.common.StyxFutures
+import com.hotels.styx.infrastructure.configuration.json.ObjectMappers.addStyxMixins
+import io.kotlintest.matchers.collections.shouldContain
+import io.kotlintest.matchers.collections.shouldContainAll
+import io.kotlintest.specs.DescribeSpec
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import java.io.File
+import java.nio.file.Paths
+import java.util.concurrent.TimeUnit.MILLISECONDS
+import java.util.concurrent.TimeUnit.SECONDS
+
+
+class FileBackedBackendServicesRegistryKotlinTest: DescribeSpec() {
+    val Mapper = addStyxMixins(ObjectMapper(YAMLFactory()))
+
+    val backendServiceOne = BackendService.Builder()
+            .id("webapp")
+            .origins(origin("webapp", "webapp-02", "localhost", 9091), origin("webapp", "webapp-01", "localhost", 9090))
+            .connectionPoolConfig(ConnectionPoolSettings.Builder()
+                    .connectTimeout(8000, MILLISECONDS)
+                    .maxConnectionsPerHost(300)
+                    .maxPendingConnectionsPerHost(50)
+                    .build())
+            .healthCheckConfig(newHealthCheckConfigBuilder()
+                    .uri("/version.txt")
+                    .interval(23456)
+                    .timeout(2000)
+                    .healthyThreshold(3)
+                    .unhealthyThreshold(5)
+                    .build())
+            .stickySessionConfig(StickySessionConfig.Builder()
+                    .enabled(true)
+                    .timeout(86400, SECONDS)
+                    .build())
+            .build()
+
+
+    fun origin(appId: String = "", id: String, host: String, port: Int): Origin {
+        return newOriginBuilder(host, port)
+                .applicationId(appId)
+                .id(id)
+                .build()
+    }
+
+    fun writeToFile(applications: List<BackendService>, path: String): File {
+        val output = File(createOnMissing(path))
+        Mapper.writer().writeValue(output, applications)
+        return output
+    }
+
+    fun createOnMissing(path: String): String {
+        val absolutePath = Paths.get(javaClass.getResource("/").toURI()).resolve(path).toFile()
```

Comment:
Why are you using both java.io and java.io here? is it because writeValue() above require a File?

Reply:
This is also inherited from the original tests.